### PR TITLE
fix(ext/fetch): subview Uint8Array in Req/Resp

### LINKED ISSUE #18857: Response constructor not handling Uint8Array with byteOffset correctly
Code:

```js
let b = new Uint8Array(24);
b.set(new TextEncoder().encode("Hello World!"), 12);
await new Response(new Uint8Array(b.buffer, 12, 12)).text()
```

Chrome: 
Screenshot 2023-04-26 at 22 37 50

Deno:
Screenshot 2023-04-26 at 22 38 38

Might have been introduced in 1.32.(3|4|5), didn't see this in 1.32.2

### PR DESCRIPTION
**Pull request by lucacasonato**
Fixes #18857

### CODE REVIEW

**Review by ry — APPROVED** [review]
RSLGTM

**Review by bartlomieju — APPROVED** [review]
LGTM

File: ext/fetch/22_body.js [hunk 1]
```diff
@@ -394,44 +393,27 @@ function extractBody(object) {
...
   } else if (ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, object)) {
-    const copy = TypedArrayPrototypeSlice(
-      new Uint8Array(
-        object,
-        0,
-        ArrayBufferPrototypeGetByteLength(object),
-      ),
-    );
-    source = copy;
+    source = TypedArrayPrototypeSlice(new Uint8Array(object));
```
Comment (satyarohith) [inline ext/fetch/22_body.js:416]: Should we pass the length of the object for optimization?

Comment (lucacasonato) [reply ext/fetch/22_body.js:416]: No. V8 can very likely get the buffer length much faster internally than we can through JS.

**Comment by bartlomieju** [comment]
Fixes https://github.com/denoland/deno/issues/18857?

**Pull request merged**